src/functions.jl

```diff
@@ -794,6 +871,50 @@ function Base.convert(
     )
 end
 
+# ScalarNonlinearFunction
+
+function Base.convert(
```

Comment:
Just for my own future reference, how do you think we can allow the user to convert in the opposite directions, e.g. SNF to SAF (which can potentially fail of course). I imagine this would be similar for e.g. VAF to VV, SQF to SAF. Could we implement these as `convert`s with errors, or do we need another interface like bridges?

Reply:
You can add `convert` and throw an `InexactError`. MOI does this in places.

```julia
julia> convert(Int, 1.5)
ERROR: InexactError: Int64(1.5)
Stacktrace:
 [1] Int64
   @ ./float.jl:723 [inlined]
 [2] convert(#unused#::Type{Int64}, x::Float64)
   @ Base ./number.jl:7
 [3] top-level scope
   @ REPL[1]:1

help?> InexactError
search: InexactError

  InexactError(name::Symbol, T, val)

  Cannot exactly convert val to type T in a method of function name.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> convert(Float64, 1+2im)
  ERROR: InexactError: Float64(1 + 2im)
  Stacktrace:
  [...]
```

Reply:
Ah ok, I'd be interested in seeing these convert methods added in future. I may be able to contribute.